addons/necropost-highlighter/userscript.js

```diff
@@ -0,0 +1,333 @@
+    // console.log("theForum: " + theForum);
+    for (let i = 0; i < possibleTopicCells.length; i++) {
+      if (forumOrSearchPageName.includes(searchResultsPageName)) {
+        // the cell to the right of the topic cell lists the forum, in search results
+        if (!possibleTopicCells[i].nextElementSibling) {
+          // synthetic "Page n" rows attached by Infinite Scrolling on search pages
+          // have no sibling, because they span all 4 columns
+          continue;
+        }
+        theForum = possibleTopicCells[i].nextElementSibling.innerText;
+      }
+      const theTopicId = await extractTopicIdFrom(possibleTopicCells[i]);
+      if (theTopicId != 0) {
+        // No restoreCell property at this point. Only added if the cell is modified
+        const topic = {
+          topicId: theTopicId,
+          topicCell: possibleTopicCells[i],
+          forum: theForum,
+        };
+        topics.push(topic);
+        // console.log("gatherTopics: " + topic.topicId + " in " + topic.forum);
+      }
+    }
+  }
+
+  /**
+   * How many topics since the earliest on the same page to be considered a "necropost"
+   */
+  function determineTopicInterval() {
+    if (!addon.settings.get("chooseCustomTopicInterval")) {
+      return defaultTopicInterval;
+    }
+    return addon.settings.get("customTopicInterval");
+  }
+
+  /**
+   * Restores topic cells to the initial state they had before any highlighting on this page.
+   * Clears any previously gathered topics.
+   * Does not conflict with Addon: Website dark mode and customizable colors.
+   */
+  async function removeAnyPriorHighlights() {
+    topics.forEach((topic) => {
+      if (topic.restoreCell) {
+        topic.topicCell.replaceWith(topic.restoreCell);
+      }
+    });
+    topics = [];
+  }
+
+  function determineActivatedForums() {
+    if (!addon.settings.get("chooseCustomForums")) {
+      return defaultForums;
+    }
+    let userSelectedForums = [];
+    for (const setting in customForumSettingToForum) {
+      if (addon.settings.get(setting)) {
+        userSelectedForums.push(customForumSettingToForum[setting]);
+      }
+    }
+    return userSelectedForums;
+  }
+
+  /**
+   * Name of a forum this addon might process, or "Search Results" or "".
+   */
+  async function determineForumOrSearchPageName() {
+    let title = document.querySelector("title");
+    if (!title) {
+      return "";
+    }
+    let titleText = title.innerText;
+    if (titleText.includes(searchResultsPageName)) {
+      return searchResultsPageName;
+    }
+    // check each forum we explicitly know about, so might be highlighting
+    for (let setting in customForumSettingToForum) {
+      let forum = customForumSettingToForum[setting];
+      if (titleText.includes(forum)) {
+        return forum;
+      }
+    }
+    // a page we don't know how to highlight (and probably shouldn't have run on)
+    return "";
+  }
```

Comment:
No need for async here!
```suggestion
  function determineForumOrSearchPageName() {
    const title = document.querySelector("title");
    if (!title) {
      return "";
    }
    const titleText = title.innerText;
    if (titleText.includes(searchResultsPageName)) {
      return searchResultsPageName;
    }
    // check each forum we explicitly know about, so might be highlighting
    for (let setting in customForumSettingToForum) {
      let forum = customForumSettingToForum[setting];
      if (titleText.includes(forum)) {
        return forum;
      }
    }
    // a page we don't know how to highlight (and probably shouldn't have run on)
    return "";
  }
```